test/functional/tests/fault_injection/test_fault_injection_many_to_one.py

```diff
@@ -0,0 +1,258 @@
+
+    with TestRun.step("Add the first cache's exported object as a core "
+                      "to the second cache instance."):
+        cache2.add_core(core1)
+
+    with TestRun.step("Try to remove core from the first level cache."):
+        output = TestRun.executor.run_expect_fail(cli.remove_core_cmd(cache_id=str(cache1.cache_id),
+                                                                      core_id=str(core1.core_id),
+                                                                      force=True))
+        cli_messages.check_msg(output, cli_messages.remove_multilevel_core)
+
+    with TestRun.step("Stop all caches."):
+        casadm.stop_all_caches()
+
+
+@pytest.mark.parametrize("cache_mode", [CacheMode.WO, CacheMode.WT, CacheMode.WB, CacheMode.WA])
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_one_core_remove(cache_mode):
+    """
+        title: Test if OpenCAS correctly handles removal of one of multiple core devices.
+        description: |
+          When one core device is removed from a cache instance all blocks previously occupied
+          by the data from that core device should be removed. That means that the number of free
+          cache blocks should increase by the number of removed blocks.
+          Test is without pass through mode.
+        pass_criteria:
+          - No system crash.
+          - The remaining core is able to use cache.
+          - Removing core frees cache blocks occupied by this core.
+    """
+    with TestRun.step("Prepare one device for cache and two for core."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(512, Unit.MebiByte)])
+        cache_dev = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(1, Unit.GibiByte)] * 2)
+        core_part1 = core_dev.partitions[0]
+        core_part2 = core_dev.partitions[1]
+
+    with TestRun.step("Start cache"):
+        cache = casadm.start_cache(cache_dev, cache_mode, force=True)
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+
+    with TestRun.step("Add both core devices to cache."):
+        core1 = cache.add_core(core_part1)
+        core2 = cache.add_core(core_part2)
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != 2:
+            TestRun.fail(f"Expected cores count: 2; Actual cores count: {cores_count}.")
+
+    with TestRun.step("Fill cache with pages from the first core."):
+        blocks = int(cache.size.value / block_size.value)
+        dd = (Dd()
+              .input(core1.system_path)
+              .output("/dev/null")
+              .block_size(block_size)
+              .count(blocks))
+        dd.run()
```

Comment:
This will not work for WO cache mode, because other cache modes insert reads to cache and write-only mode does not. It only caches writes to exported object, not reads from it. I suggest change the @pytest.mark.parametrize("cache_mode", [CacheMode.WO, CacheMode.WT, CacheMode.WB, CacheMode.WA])

to:
@pytest.mark.parametrize("cache_mode", [mode for mode in CacheMode if
                                        CacheModeTrait.InsertRead & CacheMode.get_traits(mode)])

or make an 'if condition' like you did in the test below